packages/destination-actions/src/destinations/stackadapt/forwardEvent/index.ts

```diff
@@ -0,0 +1,288 @@
+import type { ActionDefinition } from '@segment/actions-core'
+import type { Settings } from '../generated-types'
+import type { Payload } from './generated-types'
+import isEmpty from 'lodash/isEmpty'
+import isEqual from 'lodash/isEqual'
+
+const action: ActionDefinition<Settings, Payload> = {
+  title: 'Forward Event',
+  description: 'Forward Segment events to StackAdapt for conversion tracking',
+  defaultSubscription: 'type = "identify" or type = "page" or type = "screen" or type = "track"',
+  fields: {
+    user_id: {
+      label: 'Segment User ID',
+      description: 'The ID of the user in Segment',
+      type: 'string',
+      required: true,
+      default: {
+        // By default we want to use the permanent user id that's consistent across a customer's lifetime.
+        // But if we don't have that we can fall back to the anonymous id
+        '@if': {
+          exists: { '@path': '$.userId' },
+          then: { '@path': '$.userId' },
+          else: { '@path': '$.anonymousId' }
+        }
+      }
+    },
+    event_type: {
+      label: 'Event Type',
+      description: 'The Segment event type (page, track, etc.)',
+      type: 'string',
+      default: {
+        '@path': '$.type'
+      }
+    },
+    ip_fwd: {
+      description: 'IP address of the user',
+      label: 'IP Address',
+      type: 'string',
+      default: {
+        '@path': '$.context.ip'
+      }
+    },
+    title: {
+      type: 'string',
+      description: 'The title of the page where the event occurred.',
+      label: 'Page Title',
+      default: { '@path': '$.context.page.title' }
+    },
+    url: {
+      type: 'string',
+      description: 'The URL of the page where the event occurred.',
+      label: 'URL',
+      default: { '@path': '$.context.page.url' }
+    },
+    referrer: {
+      type: 'string',
+      description: 'The referrer of the page where the event occurred.',
+      label: 'Referrer',
+      default: { '@path': '$.context.page.referrer' }
+    },
+    utm_source: {
+      type: 'string',
+      format: 'text',
+      label: 'UTM Source',
+      description: 'UTM source parameter associated with event',
+      default: { '@path': '$.context.campaign.source' }
+    },
+    user_agent: {
+      description: 'User-Agent of the user',
+      label: 'User Agent',
+      type: 'string',
+      default: {
+        '@path': '$.context.userAgent'
+      }
+    },
+    email: {
+      label: 'Email',
+      description: 'Email address of the individual who triggered the event.',
+      type: 'string',
+      format: 'email',
+      default: {
+        '@if': {
+          exists: { '@path': '$.traits.email' },
+          then: { '@path': '$.traits.email' },
+          else: { '@path': '$.properties.email' }
```

Comment:
For PII fields we fallback to trying to load them from `properties` when they are not present in `traits` - this is different from the destination function code that only took them from `traits`. Let me know if there's a particular reason to read from `traits` only.